File: server/utils/getGrade.go
package utils

var grades map[string]int

func GetGrade(grade string) int {
	grades = map[string]int{
		"A+": 9,
		"A": 8,
		"A-": 7,
		"B": 6,
		"C": 5,
		"D": 4,
		"E": 3,
		"F": 2,
		"M": 1,
		"T": 0,
	}
	resp, ok := grades[grade]
	if !ok {
		return -1
	}
	return resp
}

File: frontend/src/utils/getGrade.js
const variants = new Map()

variants.set("A+", "success")
variants.set("A", "success")
variants.set("A-", "success")
variants.set("B", "warning")
variants.set("C", "warning")
variants.set("D", "warning")
variants.set("F", "danger")
variants.set("M", "danger")
variants.set("T","danger")

function getGrade(grade) {

    return variants.get(grade) || 'danger'
}

export default getGrade